lectures/supply_demand_heterogeneity.md

```diff
@@ -256,64 +202,68 @@ class ExchangeEconomy:
             if sum(Ws) != 0:
                 raise Exception('invalid wealth distribution')
 
-        self.Pi, self.bs, self.es, self.Ws, self.n, self.m = Pi, bs, es, Ws, n, m
+        self.Π, self.bs, self.es, self.Ws, self.n, self.m = Π, bs, es, Ws, n, m
 
     def competitive_equilibrium(self):
         """
         Compute the competitive equilibrium prices and allocation
         """
-        Pi, bs, es, Ws = self.Pi, self.bs, self.es, self.Ws
+        Π, bs, es, Ws = self.Π, self.bs, self.es, self.Ws
         n, m = self.n, self.m
-        slope_dc = inv(Pi.T @ Pi)
-        Pi_inv = inv(Pi)
+        slope_dc = inv(Π.T @ Π)
+        Π_inv = inv(Π)
 
         # aggregate
         b = sum(bs)
         e = sum(es)
 
         # compute price vector with mu=1 and renormalize
-        p = Pi.T @ b - Pi.T @ Pi @ e
+        p = Π.T @ b - Π.T @ Π @ e
         p = p / p[0]
 
-        # compute marg util of wealth
-        mu_s = []
+        # compute marginal utility of wealth
+        μ_s = []
         c_s = []
         A = p.T @ slope_dc @ p
 
         for i in range(m):
-            mu_i = (-Ws[i] + p.T @ (Pi_inv @ bs[i] - es[i])) / A
-            c_i = Pi_inv @ bs[i] - mu_i * slope_dc @ p
-            mu_s.append(mu_i)
+            μ_i = (-Ws[i] + p.T @ (Π_inv @ bs[i] - es[i])) / A
+            c_i = Π_inv @ bs[i] - μ_i * slope_dc @ p
+            μ_s.append(μ_i)
             c_s.append(c_i)
 
         for c_i in c_s:
             if any(c_i < 0):
                 print('allocation: ', c_s)
                 raise Exception('negative allocation: equilibrium does not exist')
 
-        return p, c_s, mu_s
+        return p, c_s, μ_s
 ```
 
-### Example: Two-Person Economy **without** Production
-  * Study how competitive equilibrium $p, c^1, c^2$ respond to  different
+## Implementation
+
+Next we use the class ``ExchangeEconomy`` defined above to study 
 
-     * $b^i$'s
-     * $e^i$'s
+* a two-person economy without production,
+* a dynamic economy, and
+* an economy with risk and arrow securities.
 
+### Two-person economy without production
 
+Here we study how competitive equilibrium $p, c_1, c_2$ respond to different $b_i$ and $e_i$, $i \in \{1, 2\}$.
```

Comment:
Change $c^1, c^2$ and $b^i$ and $e^i$ to $c_1, c_2$ and $b_i$ and $e_i$ aligned with math notations used above.